src/spi.rs

```diff
@@ -35,6 +35,7 @@ use core::cell::UnsafeCell;
 use core::cmp::{max, min, Ordering};
 use core::marker::PhantomData;
 use core::ptr;
+use std::borrow::Cow;
```

Comment:
`Cow` means at least `alloc` and so far we were able to avoid it in `esp-idf-hal` except for extra APIs.

Reply:
Yeah I'll have to re-implement `Cow`.

Reply:
You are over-doing it. If it is about ownership, just stuff in-there a `owned: bool` or a similar thing and be done with it! :)

Reply:
It bothers me but if you're fine with it then a simple `bool` will do then.

Reply:
Simple `bool` then.